pkg/workerapi/workerserver/artifactreader.go

```diff
@@ -0,0 +1,12 @@
+package workerserver
+
+import (
+	"io"
+)
+
+// ArtifactReader is an interface that provides a way to read an artifact's
+// data using the artifact's ID.
+type ArtifactReader interface {
+	// Get gets an io.ReadCloser for the artifact with the given ID.
+	Get(artifactID uint) (io.ReadCloser, error)
```

Comment:
Given the interface name, the method should conventionally be named `ReadArtifact` or something similar :)

Also, refrain from using only `Get` or such broad terms in method names as they make it so you cannot have a type that implements multiple interfaces, as Go doesn't support method overloading. It collides too easily with other interfaces.

Would be fine if it was more generic though, such as:

```go
type Getter interface {
    Get(id uint) (io.ReadCloser, error)
}
```

Or maybe more specifically:

```go
type ArtifactFileOpener interface {
    OpenArtifactFile(artifactID uint) (io.ReadCloser, error)
}
```

Reply:
I like the more generic idea, but will go with the more specific one for now while we only have this one use case 👍🏻